### PR TITLE
feat(draw): added draw component and translations

### PR DESCRIPTION
**Pull request by Unraveler**
Added Draw Tool to Tools menu:
Now it's possible to draw a point, a line, a polygon, a circle, a rectangle and write annotations.
It also possible to modify these features, upload new ones through geojson files or delete the existing ones

Features are are not persistent.

Please review @terrestris/devs

### CODE REVIEW

**Review by dnlkoch — CHANGES_REQUESTED** [review]
Some files seem to be missing in the PR.

**Comment by Unraveler** [comment]
Added missing files

File: src/components/ToolMenu/Draw/index.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+
+import {
+  faDrawPolygon,
+  faPenRuler,
+  faCircle,
+  faFont,
+  faSquare
+
```
Comment (dnlkoch) [inline src/components/ToolMenu/Draw/index.tsx:9]:
```suggestion
```

File: src/components/ToolMenu/Draw/index.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  return (
+    <ToggleGroup>
+      <DrawButton
+        name="drawPoint"
+        drawType="Point"
+      >
+        <FontAwesomeIcon icon={faCircle} />
+        <span className="draw-text">{t('Draw.point')}</span>
+      </DrawButton>
+      <DrawButton
```
Comment (dnlkoch) [inline src/components/ToolMenu/Draw/index.tsx:52]:
```suggestion

      <DrawButton
```

**Review by dnlkoch — COMMENTED** [review]
Very nice @Unraveler!

Can we also include a button for modifying an existing geometry? And would it be possible to create a import/export functionality (e.g. via a GeoJSON file)?

File: src/components/ToolMenu/Draw/index.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      </DrawButton>
+
+      <ModifyButton>
+        <FontAwesomeIcon icon={faPenToSquare} />
+        <span className="draw-modify">{t('Draw.modify')}</span>
+      </ModifyButton>
+
+      <UploadButton
+        onChange={e => {
+          // @ts-ignore
```
Comment (dnlkoch) [inline src/components/ToolMenu/Draw/index.tsx:138]: Any chance to get rid of this?

File: src/components/ToolMenu/Draw/index.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+export interface DrawProps extends Partial<DefaultDrawProps> { }
+
+export const Draw: React.FC<DrawProps> = (): JSX.Element => {
+  const {
+    t
+  } = useTranslation();
+
+  const map = useMap();
+
+  const onGeoJSONUpload = (geoJSONFile: any ) => {
```
Comment (dnlkoch) [inline src/components/ToolMenu/Draw/index.tsx:50]: Please specify the correct type.

File: src/components/ToolMenu/Draw/index.tsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    // console.log(geoJSONFile);
+    const type = geoJSONFile.type !== 'application/geojson' ? geoJSONFile.type : 'NOT SUPPORTED';
```
Comment (dnlkoch) [inline src/components/ToolMenu/Draw/index.tsx:52]: Looks quite complicated to me, we might want to check for the supported type only an return if an unsupported one has been handed over.

File: src/components/ToolMenu/Draw/index.tsx [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+          const digitizeLayer = DigitizeUtil.getDigitizeLayer(map);
+          const digitizeLayerSource = digitizeLayer.getSource();
+          digitizeLayerSource?.addFeatures(geoJSONFeatures);
+        }
+
+      };
+
+      fileReader.readAsText(geoJSONFile);
+    } else {
+      alert('NOT SUPPORTED');
```
Comment (dnlkoch) [inline src/components/ToolMenu/Draw/index.tsx:73]: See comment above.

File: src/components/ToolMenu/Draw/index.tsx [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+  } = useTranslation();
+
+  const map = useMap();
+
+  const onGeoJSONUpload = (geoJSONFile: any ) => {
+    // console.log(geoJSONFile);
+    const type = geoJSONFile.type !== 'application/geojson' ? geoJSONFile.type : 'NOT SUPPORTED';
+    if (type !== 'NOT SUPPORTED') {
+
+      var fileReader=new FileReader();
```
Comment (dnlkoch) [inline src/components/ToolMenu/Draw/index.tsx:55]:
```suggestion
      const fileReader = new FileReader();
```

File: src/components/ToolMenu/Draw/index.tsx [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+
+      fileReader.onload = () => {
+        const geoJSONFeatures = new GeoJSON().readFeatures(fileReader.result);
+        geoJSONFeatures.forEach(feat => {
+          // TODO: Add some rule to have this projection always 4326 or 3857
+          feat.getGeometry()?.transform('EPSG:4326', 'EPSG:3857');
```
Comment (dnlkoch) [inline src/components/ToolMenu/Draw/index.tsx:61]: GeoJSON should be EPSG:4326 always and we nigjt want to get the current projection from the map/view here.

File: src/components/ToolMenu/Draw/index.tsx [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+
+export const Draw: React.FC<DrawProps> = (): JSX.Element => {
+  const {
+    t
+  } = useTranslation();
+
+  const map = useMap();
+
+  const onGeoJSONUpload = (geoJSONFile: File) => {
+    var fileReader=new FileReader();
```
Comment (LukasLohoff) [inline src/components/ToolMenu/Draw/index.tsx:53]:
```suggestion
    const fileReader = new FileReader();
```

**Comment by LukasLohoff** [comment]
Looks good! Left some minor comments together with @dnlkoch 

What do you think about an export button? maybe that's an option for a follow-up MR.

**Review by LukasLohoff — APPROVED** [review]

**Pull request merged**